tests/models-response-based/src/commonTest/kotlin/test/BasicTest.kt

```diff
@@ -105,4 +106,12 @@ class BasicTest {
     }
     assertEquals(name, "R2-D2")
   }
+
+  @Test
+  fun equalsAndHashCodeAreCorrect() = runTest {
```

Comment:
FWIW, `model-xyz-based` tests should ideally be only be about differences between codegens. 

Since this doesn't depend on the codegen, I was going to say to move it to `integration-tests` but that won't work either because `integration-tests` are also run in Java. So long story short LGTM 👍. (And life is hard 😅)

Reply:
`integration-tests` is:

* JS (with Java codegen)
* Native (with Kotlin codegen)
* JVM (with Kotlin codegen and Java codegen)

See [here](https://github.com/apollographql/apollo-kotlin/blob/09f4a04a9609b609d9c011b267a7b767309fe244/tests/integration-tests/build.gradle.kts#L10) for the setup.

The nice thing is that it makes sure the same callsites work exactly the same with both Kotlin and Java models 😎 

The only thing that doesn't work is default arguments and named arguments because they do not exist in Java. For this we have a specific module named [`kotlin-codegen`](https://github.com/apollographql/apollo-kotlin/tree/09e654447c87dd8c4177b796c590878ce8ee380d/tests/kotlin-codegen)

Reply:
If it works in Java, it might be worth moving the test to `integration-test` then, increases the test coverage and makes more sense from a separation of concernes point of view